lib/tests/formulae_dependents.rb

```diff
@@ -0,0 +1,179 @@
+# frozen_string_literal: true
+
+module Homebrew
+  module Tests
+    class FormulaeDependents < TestFormulae
+      def run!(args:)
+        each_formulae.each do |f|
+          dependents!(f, args: args)
+        end
+      end
+
+      def dependents!(formula_name, args:)
+        test_header(:FormulaeDependents, method: "dependents!(#{formula_name})")
+
+        @built_formulae ||= []
+        @built_formulae << formula_name
+
+        formula = Formulary.factory(formula_name)
+        if formula.disabled?
+          ofail "#{formula.full_name} has been disabled!"
+          skip formula.name
+          return
+        end
+        new_formula = Array(@added_formulae).include?(formula_name)
+
+        if Hardware::CPU.arm? &&
+           ENV["HOMEBREW_REQUIRE_BOTTLED_ARM"] &&
+           !formula.bottled? &&
+           !formula.bottle_unneeded? &&
+           !new_formula
+          opoo "#{formula.full_name} has not yet been bottled on ARM!"
+          skip formula.name
+          return
+        end
+
+        if OS.linux? &&
+           tap.present? &&
+           tap.full_name == "Homebrew/homebrew-core" &&
+           ENV["HOMEBREW_REQUIRE_BOTTLED_LINUX"] &&
+           !formula.bottled? &&
+           !formula.bottle_unneeded?
+          opoo "#{formula.full_name} has not yet been bottled on Linux!"
+          skip formula.name
+          return
+        end
+
+        return unless satisfied_requirements?(formula, :stable)
+
+        @source_dependents.each do |dependent|
+          install_dependent_from_source(dependent, args: args)
+
+          bottled = with_env(HOMEBREW_SKIP_OR_LATER_BOTTLES: "1") do
+            dependent.bottled?
+          end
+          install_bottled_dependent(dependent, args: args) if bottled
+        end
+
+        @bottled_dependents.each do |dependent|
+          install_bottled_dependent(dependent, args: args)
+        end
+      ensure
+        cleanup_bottle_etc_var(formula) if args.cleanup?
+
+        test "brew", "uninstall", "--force", *@unchanged_dependencies if @unchanged_dependencies.present?
```

Comment:
`cleanup_bottle_etc_var` definitely needs to be in `FormulaeBottle`. Doesn't matter much for dependents.

The uninstall step should also be there too. It does mean having to reinstall the dependencies when testing dependents, which is a little bit of a slowdown, but the uninstall is important to prevent polluting other bottle builds.